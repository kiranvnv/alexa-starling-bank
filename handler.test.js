const lodash = require('lodash');

const handler = require('./handler');

describe('handle', () => {
  describe('with STARLING_ACCESS_TOKEN set', () => {
    // TODO: Add an integration test
  });

  describe('with no STARLING_ACCESS_TOKEN set', () => {
    const originalEnv = lodash.cloneDeep(process.env);

    beforeEach(() => {
      delete process.env.STARLING_ACCESS_TOKEN;
    });

    afterEach(() => {
      process.env = originalEnv;
    });

    it('returns a valid Alexa response telling you to set the token', () => {
      const mockCallback = jest.fn().mockImplementation((_, response) => {
        expect(response).toMatchSnapshot();
      });

      handler.handle({}, {}, mockCallback);
    });
  });
});

