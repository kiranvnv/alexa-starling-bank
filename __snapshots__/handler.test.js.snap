// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handle with no STARLING_ACCESS_TOKEN set returns a valid Alexa response telling you to set the token 1`] = `
Object {
  "response": Object {
    "outputSpeech": Object {
      "text": "You haven't set a Starling access token. Set the environment variable as described in the readme, re-deploy, and then try again.",
      "type": "PlainText",
    },
    "shouldEndSession": true,
  },
  "version": "1.0",
}
`;
